lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -173,21 +175,36 @@ def check_valid_modification(module, values, modifiable_params):
             module.fail_json(msg="%s is not a modifiable parameter. Valid parameters to modify are: %s." % (parameter, modifiable_params.keys()))
 
         # check allowed datatype for modified parameters
-        str_to_type = {"integer": int, "string": text_type}
-        if not isinstance(new_value, str_to_type[modifiable_params[parameter][1]]):
-            module.fail_json(msg="%s (type %s) is not an allowed value for the parameter %s. Expected a type %s." %
-                             (new_value, type(new_value), parameter, modifiable_params[parameter][1]))
+        str_to_type = {"integer": int, "string": str}
```

Comment:
I don't think this will work the way you expect with Python 2/3. I don't *think* we'll ever get `unicode` inputs, but if someone did a `!!python/unicode sometext` in their play it would show as unicode.